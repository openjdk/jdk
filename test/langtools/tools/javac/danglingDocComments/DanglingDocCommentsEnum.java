/*
 * @test /nodynamiccopyright/
 * @compile -Xlint:dangling-doc-comments DanglingDocCommentsEnum.java
 * @compile/ref=empty.out -XDrawDiagnostics DanglingDocCommentsEnum.java
 * @compile/ref=DanglingDocCommentsEnum.enabled.out -XDrawDiagnostics -Xlint:dangling-doc-comments DanglingDocCommentsEnum.java
 */

// This is a test of duplicate and misplaced doc comments in an enum class, using traditional comments

/** Bad/Extra Enum Comment. */
/** Good Enum Comment. */
@Deprecated
/** Misplaced: after anno. */
public /** Misplaced: after mods. */ enum DanglingDocCommentsEnum /** Misplaced: after ident */
{
    /** Bad/Extra enum-member Comment. */
    /**
     * Good enum-member Comment.
     */
    E1;

    /** Bad/Extra Field Comment. */
    /**
     * Good Field Comment.
     */
    public int i;

    /** Bad/Extra Method Comment. */
    /**
     * Good Method Comment.
     */
    public void m1() {
    }

    @SuppressWarnings("dangling-doc-comments")
    /** Bad/misplaced/suppressed comment. */
    public void m2() {
    }
}