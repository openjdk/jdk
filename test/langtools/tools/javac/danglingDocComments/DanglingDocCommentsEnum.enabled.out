DanglingDocCommentsEnum.java:10:1: compiler.warn.dangling.doc.comment
DanglingDocCommentsEnum.java:13:1: compiler.warn.dangling.doc.comment
DanglingDocCommentsEnum.java:14:8: compiler.warn.dangling.doc.comment
DanglingDocCommentsEnum.java:14:67: compiler.warn.dangling.doc.comment
DanglingDocCommentsEnum.java:16:5: compiler.warn.dangling.doc.comment
DanglingDocCommentsEnum.java:22:5: compiler.warn.dangling.doc.comment
DanglingDocCommentsEnum.java:28:5: compiler.warn.dangling.doc.comment
7 warnings
